### PR TITLE
Minor: add editorconfig

### PR DESCRIPTION
**Pull request by jackwener**
It's a small but useful change.

It's will enable `new_line in end` which is good for `code-style`.

set `end_of_line = lf` is useful to the developers using Windows.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+
+[*.{java,xml,py}]
+indent_style = space
+indent_size = 4
```
Comment (imbajin) [inline .editorconfig:27]: add a new line in the end of file, and When and where will this file take effect? (only in win or some IDE?)

Comment (jackwener) [reply .editorconfig:27]: https://editorconfig.org/

`editorconfig` is just work in almost IDE(vsc, IDEA.....) .

Comment (imbajin) [reply .editorconfig:27]: ok, seems fine

should we add more options? @seagle-yuan

Comment (javeme) [inline .editorconfig:27]: expect a blank line at end of ".editorconfig" file

**Review by javeme — DISMISSED** [review]
LGTM! Thanks for your contribution

File: .editorconfig (hunk 1 above)
Comment (jackwener) [reply .editorconfig]: fixed

**Review by javeme — APPROVED** [review]

**Pull request merged**